pages/prep-and-follow-up.js

```diff
@@ -0,0 +1,112 @@
+import { warmUpCoolDownPrograms } from "@/lib/warmupCooldownPrograms";
+import { warmUpCoolDownExercises } from "@/lib/warmUpCoolDownExercises";
+import {
+  ProgramCard,
+  StyledTable,
+  StyledTableRow,
+  StyledTableHeader,
+  StyledTableData,
+  StyledTableDataExercises,
+  StyledInfoButton,
+  InstructionItem,
+  InstructionsList,
+  StyledHeadline,
+  StyledCardHeadlineCoolDown,
+} from "@/styledComponents";
+import { useState } from "react";
+import React from "react";
+
+function findWarmupCooldownExerciseById(exerciseId) {
+  const stringId = String(exerciseId);
+  const foundExercise = warmUpCoolDownExercises.find(
+    (exercise) => exercise.id === stringId
+  );
+  if (!foundExercise) {
+    console.warn(`Exercise with id ${stringId} not found`);
+  }
+  return foundExercise;
+}
+
+export default function PrepAndFollowUp() {
+  const [showDetails, setShowDetails] = useState({});
+
+  const toggleDetails = (exerciseId) => {
+    setShowDetails((prev) => ({
+      ...prev,
+      [exerciseId]: !prev[exerciseId],
+    }));
+  };
+
+  return (
+    <div>
+      <StyledHeadline>Heat It Up & Chill It Out</StyledHeadline>
+      <div
+        style={{ display: "flex", flexWrap: "wrap", justifyContent: "center" }}
+      >
+        {warmUpCoolDownPrograms.map((program) => (
+          <ProgramCard key={program.id}>
+            <StyledCardHeadlineCoolDown>
+              {program.name}
+            </StyledCardHeadlineCoolDown>
+            <StyledTable>
+              <thead>
+                <StyledTableRow>
+                  <StyledTableHeader>Exercise</StyledTableHeader>
+                  <StyledTableHeader>Reps</StyledTableHeader>
+                  <StyledTableHeader>Duration</StyledTableHeader>
+                </StyledTableRow>
+              </thead>
+              <tbody>
+                {program.exerciseIds.map((exerciseId) => {
+                  const exercise = findWarmupCooldownExerciseById(exerciseId);
+
+                  if (!exercise) {
+                    return null;
+                  }
+                  const isDetailsVisible = showDetails[exercise.id] || false;
```

Comment:
i would recommend using another id here as using exercise.id leads to the problem that all exercise details in different warm-ups are toggled, which looks a bit buggy epecially in desktop view

Reply:
Thanks!
We just structered the data-lib new.